bin/metababel

```diff
@@ -122,44 +103,51 @@ end
 # Indeeed the <%= body %> will be indented, but we don't don't want it,
 # in the body string is already indented
 # But we clean the white space empty line afterward \o/
-def wrote_event_dispatchers(folder, t, matching_dispatch_data, callback_types)
+def wrote_event_dispatchers(folder, trace_model, trace_matcher, callback_types)
   event_name = 'event'
 
-  dispatchers = t.map_event_classes_with_index do |e, index_stream_class, index_event_class, default_clock_class|
+  dispatchers = trace_model.map_event_classes_with_index do |e, index_stream_class, index_event_class, default_clock_class|
     arg_variables = {}
     body = Babeltrace2Gen.context(indent: 1) do
       e.get_getter(event: event_name, arg_variables: arg_variables)
     end
 
-    # Automatic event dispatcher
+    # automatic event dispatcher
     evt_args = []
     evt_args += [GeneratedArg.new('int64_t', '_timestamp')] if default_clock_class
     evt_args += arg_variables.fetch('outputs', [])
-    dispatch_types = [DispatchType.new(e.name_sanitized, evt_args, 'generic')]
-
-    # User event dispatchers
-    usr_evt_args = evt_args + [GeneratedArg.new('const char *', '_event_class_name')]
-    dispatch_types += matching_dispatch_data.filter_map do |md|
-      dispatcher_evt_args = btx_get_matching_args(e.name, usr_evt_args, md[:regex], md.fetch(:args, []))
-      unless dispatcher_evt_args.nil?
-        DispatchType.new(sanitize(md[:name]), dispatcher_evt_args,
-                         "matching_#{sanitize(md[:name])}")
-      end
-    end
+    dispatch_type = DispatchType.new(e.name_sanitized, evt_args, 'generic')
 
     Dispatcher.new(
       e.name, arg_variables.fetch('outputs', []),
       body, index_stream_class, index_event_class,
       default_clock_class,
       arg_variables.fetch('outputs_allocated', []),
-      dispatch_types
+      [dispatch_type],
+      e
     )
   end
 
+  trace_matcher.map_event_classes do |em|
```

Comment:
Mayve a little comment
"Matching callbacks. Mutate the dispatchera dispatch_types"